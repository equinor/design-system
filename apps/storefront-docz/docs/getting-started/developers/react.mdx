---
title: Developers
tabs:
  - Introduction
  - CSS framework
  - React
  - Figma
mode: draft
route: /getting-started/developers/react/
---

:warning: Information coming soon
