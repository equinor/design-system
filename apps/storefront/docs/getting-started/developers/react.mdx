---
title: Developers
metaTitle: Developers react
tabs:
  - Introduction
  - CSS framework
  - React
  - Figma
mode: draft
route: /getting-started/developers/react/
---

:warning: Information coming soon
