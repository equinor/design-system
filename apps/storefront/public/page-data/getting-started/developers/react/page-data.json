{"componentChunkName":"component---docs-getting-started-developers-react-mdx","path":"/getting-started/developers/react/","result":{"pageContext":{"frontmatter":{"title":"Developers","metaTitle":"Developers react","tabs":["Introduction","CSS framework","React","Figma"],"mode":"draft","route":"/getting-started/developers/react/"},"entry":{"id":"6f9784da00517d57e4627269567073e6","filepath":"docs/getting-started/developers/react.mdx","fullpath":"/Users/vnys/Kode/storefront-replace/apps/storefront/docs/getting-started/developers/react.mdx","route":"/getting-started/developers/react/","slug":"docs-getting-started-developers-react","name":"React","menu":"","headings":[]}}}}