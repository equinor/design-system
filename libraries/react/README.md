@eds-ui/react
===============

1. `npm i -g pnpm`
1. `pnpm install`
1. `pnpm recursive install`

## Run storybook

`pnpm start`

## Build modules

`pnpm recursive run build`
