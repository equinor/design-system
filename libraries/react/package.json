{
  "name": "@eds-ui/react",
  "private": true,
  "version": "0.0.1",
  "description": "",
  "main": "index.js",
  "scripts": {
    "test": "jest",
    "storybook": "start-storybook"
  },
  "keywords": [],
  "author": "",
  "license": "AGPL",
  "dependencies": {
    "react": "^16.8.6",
    "react-dom": "^16.8.6"
  },
  "devDependencies": {
    "@babel/cli": "^7.4.4",
    "@babel/core": "^7.4.4",
    "@babel/preset-env": "^7.4.4",
    "@babel/preset-react": "^7.0.0",
    "@storybook/react": "^5.0.11",
    "babel-plugin-styled-components": "^1.10.0",
    "enzyme": "^3.9.0",
    "enzyme-adapter-react-16": "^1.12.1",
    "jest": "^24.8.0",
    "jest-styled-components": "^6.3.1",
    "rollup": "^1.11.3",
    "rollup-plugin-babel": "^4.3.2",
    "rollup-plugin-commonjs": "^9.3.4",
    "rollup-plugin-node-resolve": "^4.2.3",
    "styled-components": "^4.2.0"
  },
  "peerDependencies": {
    "react": "^16.8.6",
    "react-dom": "^16.8.6",
    "styled-components": "^4.2.0",
    "prop-types": "^15.7.2"
  },
  "jest": {
    "verbose": true,
    "setupFilesAfterEnv": [
      "<rootDir>/testSetup.js"
    ],
    "testPathIgnorePatterns": [
      "<rootDir>/(node_modules)"
    ],
    "transformIgnorePatterns": [
      "<rootDir>/(node_modules)"
    ],
    "haste": {
      "providesModuleNodeModules": [
        ".*"
      ]
    },
    "roots": [
      "packages/node_modules"
    ]
  }
}
