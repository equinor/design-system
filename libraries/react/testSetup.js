import React from 'react'
import Enzyme from 'enzyme'
import 'jest-styled-components'
import styled from 'styled-components'
import Adapter from 'enzyme-adapter-react-16'

Enzyme.configure({ adapter: new Adapter() })
