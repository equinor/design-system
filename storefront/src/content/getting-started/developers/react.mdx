---
title: React
tabs:
    - Introduction
    - CSS framework
    - React
    - Figma
    - Contributing
mode: draft
---


:warning: Information coming soon
