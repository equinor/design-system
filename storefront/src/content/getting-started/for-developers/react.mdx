---
title: React
tabs:
    - Introduction
    - CSS framework
    - React
    - Figma
    - Contributing
---

Developers React
