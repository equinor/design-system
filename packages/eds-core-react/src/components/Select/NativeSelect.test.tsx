import { useState } from 'react'
import { render, fireEvent, screen, act } from '@testing-library/react'
import { axe } from 'jest-axe'
import styled from 'styled-components'
import { NativeSelect } from '.'

type SelectProps = {
  label: string
  id: string
}
const SelectMarkup = ({ label, id }: SelectProps) => {
  return (
    <NativeSelect label={label} id={id}>
      <option>Option one</option>
      <option>Option two</option>
      <option>Option three</option>
    </NativeSelect>
  )
}

describe('NativeSelect', () => {
  it('Matches snapshot', () => {
    const { asFragment } = render(
      <NativeSelect label="label" id="native-select-snapshot">
        <option>Option one</option>
        <option>Option two</option>
        <option>Option three</option>
      </NativeSelect>,
    )
    expect(asFragment()).toMatchSnapshot()
  })
  it('Matches multi-select snapshot', () => {
    const { asFragment } = render(
      <NativeSelect label="label" id="native-multi-select-snapshot" multiple>
        <option>Option one</option>
        <option>Option two</option>
        <option>Option three</option>
      </NativeSelect>,
    )
    expect(asFragment()).toMatchSnapshot()
  })
  it('Should pass a11y test', async () => {
    const { container } = render(
      <NativeSelect label="label" id="a11y-id">
        <option>Option one</option>
        <option>Option two</option>
      </NativeSelect>,
    )
    await act(async () => {
      const results = await axe(container)
      expect(results).toHaveNoViolations()
    })
  })
  it('Has provided label', () => {
    const label = 'Select label'
    render(<SelectMarkup label={label} id="test-id" />)
    const inputNode = screen.getByLabelText(label)
    expect(inputNode).toBeDefined()
  })

  it('Can be disabled', () => {
    const label = 'Select label'
    render(
      <NativeSelect label={label} id="testDisabled" disabled>
        <option>Option one</option>
        <option>Option two</option>
        <option>Option three</option>
      </NativeSelect>,
    )
    const inputNode = screen.getByLabelText(label)
    expect(inputNode).toBeDisabled()
  })

  it('Can preselect a specific option', () => {
    const label = 'Select label'
    render(
      <NativeSelect label={label} id="testPreselect" defaultValue="two">
        <option value="one">Option one</option>
        <option value="two">Option two</option>
        <option value="three">Option three</option>
      </NativeSelect>,
    )
    const inputNode = screen.getByLabelText(label)
    expect(inputNode).toHaveValue('two')
  })

  type ControlledProps = {
    onChange: () => void
  }

  const ControlledSelect = ({ onChange }: ControlledProps) => {
    const [value, setValue] = useState('one')
    return (
      <NativeSelect
        label="select-label"
        id="selectId"
        value={value}
        onChange={(e) => {
          setValue(e.target.value)
          onChange()
        }}
      >
        {' '}
        <option value="one">Option one</option>
        <option value="two">Option two</option>
        <option value="three">Option three</option>
      </NativeSelect>
    )
  }

  it('Can be a controlled component', () => {
    const newValue = 'three'
    const handleChange = jest.fn()
    render(<ControlledSelect onChange={handleChange} />)
    const select = screen.queryByLabelText('select-label')

    expect(select).toHaveValue('one')
    expect(handleChange).toHaveBeenCalledTimes(0)

    fireEvent.change(select, {
      target: { value: newValue },
    })
    expect(select).toHaveValue('three')
    expect(handleChange).toHaveBeenCalledTimes(1)
  })

  it('Can select multiple elements', () => {
    const label = 'Select label'
    render(
      <NativeSelect label={label} id="testMultiple" multiple>
        <option>Option one</option>
        <option>Option two</option>
        <option>Option three</option>
      </NativeSelect>,
    )
    const inputNode = screen.getByLabelText(label)
    expect(inputNode).toHaveAttribute('multiple')
  })

  const StyledSelect = styled(NativeSelect)`
    clip-path: unset;
  `
  it('Can extend the css for the component', () => {
    const { container } = render(<StyledSelect label="test" id="testId" />)
    // eslint-disable-next-line testing-library/no-node-access
    expect(container.firstChild).toHaveStyleRule('clip-path', 'unset')
  })
})
