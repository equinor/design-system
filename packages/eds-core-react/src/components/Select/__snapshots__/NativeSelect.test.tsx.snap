// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NativeSelect Matches multi-select snapshot 1`] = `
<DocumentFragment>
  .c1 {
  display: flex;
  justify-content: space-between;
  align-items: flex-end;
  position: relative;
  margin: 0;
  color: var(--eds_text__static_icons__tertiary, rgba(111, 111, 111, 1));
  font-family: Equinor;
  font-size: 0.750rem;
  font-weight: 500;
  line-height: 1.333em;
  text-align: left;
  margin-left: 8px;
  margin-right: 8px;
  color: var(--eds_text__static_icons__tertiary, rgba(111, 111, 111, 1));
}

.c2 {
  margin: 0;
}

.c0 {
  min-width: 100px;
  width: 100%;
}

.c3 {
  border: none;
  border-radius: 0;
  box-shadow: inset 0 -1px 0 0 var(--eds_text__static_icons__tertiary, rgba(111, 111, 111, 1));
  margin: 0;
  color: var(--eds_text__static_icons__default, rgba(61, 61, 61, 1));
  font-family: Equinor;
  font-size: 1.000rem;
  font-weight: 400;
  line-height: 1.500em;
  letter-spacing: 0.025em;
  text-align: left;
  padding-left: 8px;
  padding-top: 6px;
  padding-right: 8px;
  padding-bottom: 6px;
  padding-right: calc(8px *2 + 24px);
  display: block;
  margin: 0;
  appearance: none;
  background-image: linear-gradient( to bottom,var(--eds_ui_background__light, rgba(247, 247, 247, 1)) 0%,var(--eds_ui_background__light, rgba(247, 247, 247, 1)) 100% );
  background-repeat: no-repeat,repeat;
  background-position: right 8px top 50%;
  width: 100%;
}

.c3:active,
.c3:focus {
  box-shadow: none;
  outline: 2px solid var(--eds_interactive_primary__resting, rgba(0, 112, 121, 1));
  outline-offset: 0px;
}

.c3:disabled {
  color: var(--eds_interactive__disabled__text, rgba(190, 190, 190, 1));
  background-image: url("data:image/svg+xml,%3Csvg width='24' height='24' viewBox='0 0 24 24' xmlns='http://www.w3.org/2000/svg'%3E%3Cpath fill='%23bebebe' d='M7 9.5l5 5 5-5H7z'/%3E%3C/svg%3E"),linear-gradient( to bottom,var(--eds_ui_background__light, rgba(247, 247, 247, 1)) 0%,var(--eds_ui_background__light, rgba(247, 247, 247, 1)) 100% );
  cursor: not-allowed;
  box-shadow: none;
  outline: none;
}

.c3:disabled .arrow-icon {
  fill: red;
}

.c3:disabled:focus,
.c3:disabled:active {
  outline: none;
}

<div
    class="c0"
  >
    <label
      class="c1"
      for="native-multi-select-snapshot"
    >
      <span
        class="c2"
      >
        label
      </span>
    </label>
    <select
      class="c3"
      id="native-multi-select-snapshot"
      multiple=""
    >
      <option>
        Option one
      </option>
      <option>
        Option two
      </option>
      <option>
        Option three
      </option>
    </select>
  </div>
</DocumentFragment>
`;

exports[`NativeSelect Matches snapshot 1`] = `
<DocumentFragment>
  .c1 {
  display: flex;
  justify-content: space-between;
  align-items: flex-end;
  position: relative;
  margin: 0;
  color: var(--eds_text__static_icons__tertiary, rgba(111, 111, 111, 1));
  font-family: Equinor;
  font-size: 0.750rem;
  font-weight: 500;
  line-height: 1.333em;
  text-align: left;
  margin-left: 8px;
  margin-right: 8px;
  color: var(--eds_text__static_icons__tertiary, rgba(111, 111, 111, 1));
}

.c2 {
  margin: 0;
}

.c0 {
  min-width: 100px;
  width: 100%;
}

.c3 {
  border: none;
  border-radius: 0;
  box-shadow: inset 0 -1px 0 0 var(--eds_text__static_icons__tertiary, rgba(111, 111, 111, 1));
  margin: 0;
  color: var(--eds_text__static_icons__default, rgba(61, 61, 61, 1));
  font-family: Equinor;
  font-size: 1.000rem;
  font-weight: 400;
  line-height: 1.500em;
  letter-spacing: 0.025em;
  text-align: left;
  padding-left: 8px;
  padding-top: 6px;
  padding-right: 8px;
  padding-bottom: 6px;
  padding-right: calc(8px *2 + 24px);
  display: block;
  margin: 0;
  appearance: none;
  background-image: url("data:image/svg+xml,%3Csvg width='24' height='24' viewBox='0 0 24 24' xmlns='http://www.w3.org/2000/svg'%3E%3Cpath fill='%236f6f6f' d='M7 9.5l5 5 5-5H7z'/%3E%3C/svg%3E"),linear-gradient( to bottom,var(--eds_ui_background__light, rgba(247, 247, 247, 1)) 0%,var(--eds_ui_background__light, rgba(247, 247, 247, 1)) 100% );
  background-repeat: no-repeat,repeat;
  background-position: right 8px top 50%;
  width: 100%;
}

.c3:active,
.c3:focus {
  box-shadow: none;
  outline: 2px solid var(--eds_interactive_primary__resting, rgba(0, 112, 121, 1));
  outline-offset: 0px;
}

.c3:disabled {
  color: var(--eds_interactive__disabled__text, rgba(190, 190, 190, 1));
  background-image: url("data:image/svg+xml,%3Csvg width='24' height='24' viewBox='0 0 24 24' xmlns='http://www.w3.org/2000/svg'%3E%3Cpath fill='%23bebebe' d='M7 9.5l5 5 5-5H7z'/%3E%3C/svg%3E"),linear-gradient( to bottom,var(--eds_ui_background__light, rgba(247, 247, 247, 1)) 0%,var(--eds_ui_background__light, rgba(247, 247, 247, 1)) 100% );
  cursor: not-allowed;
  box-shadow: none;
  outline: none;
}

.c3:disabled .arrow-icon {
  fill: red;
}

.c3:disabled:focus,
.c3:disabled:active {
  outline: none;
}

<div
    class="c0"
  >
    <label
      class="c1"
      for="native-select-snapshot"
    >
      <span
        class="c2"
      >
        label
      </span>
    </label>
    <select
      class="c3"
      id="native-select-snapshot"
    >
      <option>
        Option one
      </option>
      <option>
        Option two
      </option>
      <option>
        Option three
      </option>
    </select>
  </div>
</DocumentFragment>
`;
